### PR TITLE
Complete project based on template.

### PR DESCRIPTION
**Pull request by MahdiyarGHD**

### CODE REVIEW

File: src/CSharp/EasyMicroservices.StorageMicroservice.Database/Database/Entities/FileEntity.cs [hunk 1]
```diff
@@ -5,7 +5,9 @@ namespace EasyMicroservices.StorageMicroservice.Database.Entities
     public class FileEntity : FileSchema
     {
         public long Id { get; set; }
+        public string Guid { get; set; }
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.StorageMicroservice.Database/Database/Entities/FileEntity.cs]: Change this type to System.Guid

File: src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Common/ResultContract.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    public class ResultContract
+    {
+        public ResultContract()
+        {
+            this.SetResultState(0, new object());
+        }
+        public int StatusId { get; set; }
+        public string Message { get; set; }
+        public bool IsSuccessful { get; set; }
+        public object OutputRes { get; set; }
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Common/ResultContract.cs]: Do not use the object in contract, use real type

File: src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Common/ResultContract.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            return result;
+        }
+    }
+    public class ResultContract
+    {
+        public ResultContract()
+        {
+            this.SetResultState(0, new object());
+        }
+        public int StatusId { get; set; }
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Common/ResultContract.cs]: Use Status as Enum and change its name to Status

File: src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Requests/FileRequest.cs
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Requests/FileRequest.cs:1]: Add Contract suffix end of all contracts, it must be FileRequestContract

File: src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Requests/FolderRequest.cs
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Requests/FolderRequest.cs:1]: Add Contract suffix end of all contracts, it must be FolderRequestContract

File: src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Requests/FolderRequest.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace EasyMicroservices.StorageMicroservice.Contracts
+{
+    public class AddFolderDom
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Requests/FolderRequest.cs]: Dont use the 'Dom' suffix, only Contract is good.
move it to requests.
rename it to AddFolderRequestContract

File: src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Requests/FolderRequest.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        }
+
+        [StringLength(100, MinimumLength = 5, ErrorMessage = "Folder name must be between 6 and 100 characters.")]
+        public string Name { get; set; }
+
+        [StringLength(63, MinimumLength = 5, ErrorMessage = "Folder path must be between 6 and 100 characters.")]
+        public string Path { get; set; }
+    }
+
+    public class UpdateFolderDom
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Requests/FolderRequest.cs]: Don't use the 'Dom' suffix, only Contract is good.
move it to requests.
rename it to AddFolderRequestContract

**Review by Ali-YousefiTelori — CHANGES_REQUESTED** [review]

**Review by mahdidelzendeh72 — APPROVED** [review]

File: src/CSharp/EasyMicroservices.StorageMicroservice.WebApi/Controllers/FileController.cs [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+        {
+            string webRootPath = @Directory.GetCurrentDirectory();
+            string directoryPath = _directoryManagerProvider.PathProvider.Combine(webRootPath, "wwwroot", Constants.RootAddress, FilePath); 
+            return directoryPath;
+        }
+
+
+
+        [HttpPost("AddFileAsync")]
+        public async Task<MessageContract<object>> AddSingleFileAsync([FromForm] AddFileContract input)
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.StorageMicroservice.WebApi/Controllers/FileController.cs]: Don't use object in reesult or any request

File: src/CSharp/EasyMicroservices.StorageMicroservice.WebApi/Controllers/FileController.cs [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+                }
+
+            }
+
+            return result;
+
+        }
+
+        [HttpDelete]
+        public async Task<MessageContract<FileEntity>> DeleteFileByGuidAsync(string guid, string? password)
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.StorageMicroservice.WebApi/Controllers/FileController.cs]: Don't send entities as result, only use contracts.

File: src/CSharp/EasyMicroservices.StorageMicroservice.WebApi/Controllers/FileController.cs [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+                }
+            }
+
+            return Result;
+
+        }
+
+
+        [HttpDelete]
+        public async Task<MessageContract<FileEntity>> DeleteFileByIdAsync(long id, string? password)
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.StorageMicroservice.WebApi/Controllers/FileController.cs]: don't use entities as a result only sned contracts

File: src/CSharp/EasyMicroservices.StorageMicroservice.WebApi/Controllers/FolderController.cs [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+
+        private string PathToFullPath(string name)
+        {
+            string webRootPath = @Directory.GetCurrentDirectory();
+            string directoryPath = _directoryManagerProvider.PathProvider.Combine(webRootPath, "wwwroot", Constants.RootAddress, name);
+            return directoryPath ?? _directoryManagerProvider.PathProvider.Combine(webRootPath, "wwwroot", Constants.RootAddress);
+        }
+
+        [HttpGet]
+        public async Task<MessageContract<List<FolderEntity>>> GetAll()
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.StorageMicroservice.WebApi/Controllers/FolderController.cs]: Dont use entities as service results

File: src/CSharp/EasyMicroservices.StorageMicroservice.WebApi/Controllers/FolderController.cs [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+            {
+                result.IsSuccess = false;
+                result.Error.Message = "No directory found";
+            }
+
+            return result;
+        }
+
+        [HttpPut]
+        public async Task<MessageContract<FolderEntity>> UpdateAsync(UpdateFolderContract input)
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.StorageMicroservice.WebApi/Controllers/FolderController.cs]: Dont use entities as service results

File: src/CSharp/EasyMicroservices.StorageMicroservice.WebApi/Controllers/FolderController.cs [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+            {
+                result.IsSuccess = false;
+                result.Error.Message = "Your folder path isn't valid.";
+            }
+
+            return result;
+        }
+
+        [HttpPost]
+        public async Task<MessageContract<FolderEntity>> AddAsync(AddFolderContract input)
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.StorageMicroservice.WebApi/Controllers/FolderController.cs]: Dont use entities as service results

File: src/CSharp/EasyMicroservices.StorageMicroservice.WebApi/Controllers/FolderController.cs [hunk 12]
```diff
@@ -0,0 +1,210 @@
...
+            {
+                result.IsSuccess = false;
+                result.Error.Message = "Your folder path isn't valid.";
+            }
+
+            return result;
+        }
+
+        [HttpDelete]
+        public async Task<MessageContract> DeleteAsync(long Id)
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.StorageMicroservice.WebApi/Controllers/FolderController.cs]: rename Id = fileId, don't use pascal case as parameter names

File: src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Requests/FolderRequestContract.cs
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Requests/FolderRequestContract.cs:1]: File name and class name are not same

File: src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Requests/FolderRequestContract.cs [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+        }
+
+        [StringLength(100, MinimumLength = 5, ErrorMessage = "Folder name must be between 6 and 100 characters.")]
+        public string Name { get; set; }
+
+        [StringLength(63, MinimumLength = 5, ErrorMessage = "Folder path must be between 6 and 100 characters.")]
+        public string Path { get; set; }
+    }
+
+    public class UpdateFolderContract
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Requests/FolderRequestContract.cs]: don't add two class in one file

File: src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Requests/FileRequestContract.cs
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Requests/FileRequestContract.cs:1]: File name and class name are not same

File: src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Common/ResultContract.cs
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Common/ResultContract.cs:1]: Remove this file and statuses,
MessageContract has FailedReasonType

**Review by Ali-YousefiTelori — CHANGES_REQUESTED** [review]

**Review by mahdidelzendeh72 — APPROVED** [review]

File: src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Requests/FolderRequestContract.cs
Comment (MahdiyarGHD) [reply src/CSharp/EasyMicroservices.StorageMicroservice.Domain/Contracts/Requests/FolderRequestContract.cs:1]: Just to clarify, my initial plan was to consolidate all necessary contracts for each project section into a single contract file. However, I've made the requested changes

**Review by Ali-YousefiTelori — APPROVED** [review]

**Pull request merged**